unison-share-api/src/Unison/Sync/Types.hs

```diff
@@ -110,32 +110,31 @@ instance FromJSON Base64Bytes where
 newtype RepoName = RepoName {unRepoName :: Text}
   deriving newtype (Show, Eq, Ord, ToJSON, FromJSON)
 
-data Path = Path
-  { -- This is a nonempty list, where we require the first segment to be the repo name / user name / whatever,
+data ShareLocation = ShareLocation
```

Comment:
I'm not totally sure the `ShareUserHandle` should be in the sync API types - I think we were trying to keep the API server-agnostic. You can see in the comment below that the `NonEmpty Text` may ideally be a `[Text]` to allow for servers to just present a codebase directly (without a "user" indirection). Could `Unison.Sync.Types.Path` remain as-is?

Reply:
Hrmm, I think if we want to be agnostic about server paths we would probably want to lean into that and just send a single `Text` then right?

Ultimately UCM will probably want to know about both projects and users as a matter of convenience when dealing with syncing projects, so I think we might be fooling ourselves if we try to ignore it, but could probably put it off a bit longer. Splitting up this type isn't necessary for this change, so I could revert that part, but if it's ever cast into a Unison Path it will be an invalid one, and this was already happening in the output layer, so I figured it was safest to split it into a separate type so that nobody would accidentally assume that a `Share.Path` was a valid `Unison.Path` yknow?

Reply:
Let's leave the API as `NonEmpty Text` or `[Text]` for now (nobody should accidentally assume that a `NonEmpty Text` is a valid `Unison.Path` either), and then parse that API type to some client-specific or server-specific data structure as needed.